Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1186,9 +1205,51 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     m_backgroundWebView = nil;
 }
 
+void WebExtensionContext::loadBackgroundPageListenersFromStorage()
+{
+    NSData *listenersData = [m_state objectForKey:backgroundPageListenersKey];
+    NSCountedSet *savedListeners = [NSKeyedUnarchiver unarchivedObjectOfClasses:[NSSet setWithArray:@[ NSCountedSet.class, NSNumber.class ]] fromData:listenersData error:nil];
+
+    m_backgroundPageListeners.clear();
+    for (NSNumber *entry in savedListeners)
+        m_backgroundPageListeners.add(static_cast<WebExtensionEventListenerType>(entry.unsignedIntValue), [savedListeners countForObject:entry]);
+}
+
+void WebExtensionContext::saveBackgroundPageListenersToStorage()
+{
+    if (backgroundContentURL().isEmpty() || isPersistent())
```

Comment:
Use `extension().backgroundContentIsPersistent()`. `isPersistent()` is the wrong check, that is about the extension state and storage. No need to check `backgroundContentURL()` since `backgroundContentIsPersistent` does that internally.